homeassistant/components/modbus.py

```diff
@@ -6,151 +6,192 @@
 """
 import logging
 import threading
+from typing import TYPE_CHECKING, Any, List
 
 import voluptuous as vol
 
 import homeassistant.helpers.config_validation as cv
 from homeassistant.const import (
-    EVENT_HOMEASSISTANT_START, EVENT_HOMEASSISTANT_STOP,
-    CONF_HOST, CONF_METHOD, CONF_PORT, CONF_TYPE, CONF_TIMEOUT, ATTR_STATE)
+    ATTR_STATE, CONF_HOST, CONF_METHOD, CONF_PORT, CONF_TIMEOUT, CONF_TYPE,
+    EVENT_HOMEASSISTANT_START, EVENT_HOMEASSISTANT_STOP)
 
-DOMAIN = 'modbus'
+if TYPE_CHECKING:
+    # pylint: disable=unused-import
+    from pymodbus.client.sync import BaseModbusClient
 
-REQUIREMENTS = ['pymodbus==1.3.1']
+_LOGGER = logging.getLogger(__name__)
+
+DOMAIN = "modbus"
+
+REQUIREMENTS = ["pymodbus==1.3.1"]
 
 # Type of network
-CONF_BAUDRATE = 'baudrate'
-CONF_BYTESIZE = 'bytesize'
-CONF_STOPBITS = 'stopbits'
-CONF_PARITY = 'parity'
+CONF_BAUDRATE = "baudrate"
```

Comment:
There are a lot of unrelated changes in this PR making it very hard to see what has actually changed. I'm assuming you're using some auto formatter. Please update your PR to not contain any unrelated style changes.

Reply:
I am very much interested in this, so I thought I'd help out a bit with these changes.

Also, Travis CI detected a problem in components/climate/flexit.py
I updated that to the new implementation. Unfortunately I can't really test that component.

@crhan I can't directly contribute to your PR but commits are in my own fork:
style changes in: 813d077463308ec6eab86186db751427454af2d5
flexit update in: 3149ec6b9e9f19fb3d57707b944f19eb8bd3ce22

Reply:
Thank you for this work.
I have created a new PR  #19726, so it is easier for me to make further changes.